charts/nx-agents/templates/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nx-cloud-workflow-controller
+  namespace: {{ .Values.global.namespace | quote }}
+  labels:
+    {{- include "nxCloud.app.labels" . | indent 4 }}
+  {{- if .Values.controller.deployment.annotations }}
+  annotations:
+    {{- toYaml .Values.controller.deployment.annotations | nindent 4 }}
+  {{- end
+spec:
+  selector:
+    matchLabels:
+      app: nx-cloud-workflow-controller
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: nx-cloud-workflow-controller
+    spec:
+      securityContext:
+        runAsNonRoot: true
+      containers:
+        - command:
+          - /nx-cloud-workflow-controller
+          - --namespace={{ .Values.global.namespace }}
+          - --in-cluster=true
+          {{- range $key, $value := .Values.executor.env }}
+          - --executor-env={{ $key }}={{ $value }}
+          {{- end }}
+          - --workflow-service-address=http://nx-cloud-workflow-controller-service:9000
+          image: {{ include "nxCloud.images.workflowController.image" . }}
+          imagePullPolicy: {{ .Values.controller.image.pullPolicy | quote }}
+          {{- if .Values.controller.resources }}
+          resources: {{- toYaml .Values.controller.resources | nindent 12 }}
+          {{- end }}
+          name: nx-cloud-workflow-controller
+          securityContext:
+            runAsUser: 1000
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - 'ALL'
```

Comment:
should we make this customisable? in case they want root access?

Reply:
not sure, tbh this deployment needs no extra permissions since it is just the controller, so using least-privilege model is probably best. There really shouldn't be anything it needs root for and anything else should be configured via the include RBAC